Java/TelemetrySystem/src/main/java/tddmicroexercises/solidtelemetrysystem/TelemetryDiagnosticControlsImpl.java

```diff
@@ -0,0 +1,53 @@
+package tddmicroexercises.solidtelemetrysystem;
+
+import tddmicroexercises.telemetrysystem.TelemetryClient;
+
+public class TelemetryDiagnosticControlsImpl implements ITelemetryDiagnosticControls
+{
+
+    private final String DiagnosticChannelConnectionString = "*111#";
+
+    private final TelemetryClientImpl telemetryClient;

```

Comment:
As TelemetryClientImpl implements ITelemetryClientConnection...so replace the type of telemetryClient to ITelemetryClientConnection to ensure the depdency inversion principle